platform/posix/transport/src/openssl_posix.c

```diff
@@ -581,7 +581,6 @@ OpensslStatus_t Openssl_Disconnect( NetworkContext_t * pNetworkContext )
         }
 
         SSL_free( pNetworkContext->pSsl );
-        pNetworkContext->pSsl = NULL;
```

Comment:
This was the cause of the SegFault. By setting the SSL object to NULL explicitly, the subsequent call to `OpenSSL_Send` fails. It is not required to explicitly set the SSL object member of network context to NULL. The `SSL_write` function is able to recognize if the passed SSL object has been earlier freed with `SSL_free`